index.js

```diff
@@ -21,14 +21,17 @@ var toArray = Array.from || (function (slice) {
 
 // ===================================================================
 
-function eventToPromise (emitter, event, opts) {
-  var ignoreErrors = opts && opts.ignoreErrors
+function eventToPromise (emitter, event, _opts) {
+  var opts = _opts || {}
+  var ignoreErrors = opts.ignoreErrors
+  var errorEvent = opts.error || 'error'
 
   return new Bluebird(function (resolve, reject) {
     var addListener =
+      emitter.once ||
```

Comment:
It's still needed if the error fires first, and is a no-op if the listener has already been removed.

Having said that, there's no need to remove either listener, since promises just ignore redundant resolve/reject calls. This is the basis for [this](https://github.com/rschaosid/node-event-promise) implementation.